infrastructure/web/controllers/get-admin-orders.js

```diff
@@ -1,11 +1,31 @@
 module.exports = function makeGetAdminOrders({ listOrders }) {
   return async function getAdminOrders(req, res) {
     try {
-      const { ownerId } = req.body
-      const ordersList = await listOrders({ ownerId })
-      res.render('admin/orders/index', { data: { orders: ordersList } })
+      const orderId = req.params.id
+      if (orderId) {
+        getAdminOrderById(req, res, listOrders, orderId)
+      } else {
+        // FIXME: THE USER_ID IS UNDEFINED FOR THE TIME BEIGN, SO WE ARE USING A HARD_CODED_ID
+        const user = req.user
+        const ownerId = user ? user.id : 'user_1'
+
+        const ordersList = await listOrders({ ownerId })
+        res.render('admin/orders/index', { data: { orders: ordersList } })
+      }
     } catch (err) {
       res.render('admin/orders/new', { errorMessages: [err.message] })
     }
   }
 }
+
+// NOTE: THIS METHOD CAN BE MOVED LATER TO A SEPRATE CONTROLLER
+async function getAdminOrderById(req, res, listOrders, orderId) {
```

Comment:
Yes it needs to be extracted out to a separate controller, I already did that for `get-user-product.js` controller you can take a look on it. Having that here, the getAdminOrders controller is doing too much work that it shouldn't be doing.
let's separate the concerns and have `getAdminOrders` for listing orders and `getAdminOrder` for getting a single order details.